### DISCUSSION TITLE
May I ask how Routr proxies freeswitch and how to configure it?

### DISCUSSION THREAD

**Discussion by happytulubi** [fonoster/routr#268] [Q&A]
May I ask how Routr proxies freeswitch and how to configure it?
Excuse me, how is Location Service associated with EdgePort or Dispatcher? I see that Dispatcher is configured in EdgePort configuration, but a certain processor is configured in Dispatcher configuration. Do you need to develop this processor yourself. How does Routr proxy freeswitch and configure it

**Comment by psanders** [comment]
It depends on your use case. Every case could use a different combination of components.

Some resources that might help understand each component include

- https://routr.io/docs/2.0.0/overview/concepts
- https://github.com/fonoster/routr/blob/main/CORE.md
- https://github.com/fonoster/routr/blob/main/CONNECT.md
- https://fonoster.gumroad.com/l/the-future-of-programmable-sip-servers

Quick summary:

The Location service is where Routr stores the addresses for all endpoints (almost all use cases need this)

The Dispatcher, on the other hand, directs messages to the correct feature server. It is required if you need to communicate with multiple Processors or incorporate a Middleware. Otherwise, connect the EdgePort directly to a Processor.

> Do you need to develop this processor yourself

The only Processor available now is the [Connect Processor](https://routr.io/docs/2.0.0/connect/introduction). If you have a use case outside that, you will need to build one.

**Reply by happytulubi** [reply]
What I want to ask is how to start this connect processor. I saw that the Dispatcher will configure the connect processor with port 51904
According to this link“ https://github.com/fonoster/routr/blob/main/CONNECT.md Unable to find how to start connect processor

I am now able to start edgeport, dispatcher, and location using the following commands, but I am not sure how to start this connect processor.

npm run start:edgeport
npm run start:dispatcher
npm run start:location

**Reply by psanders** [reply]
Unless you are planning on contributing, I suggest using the docker services. 

I would suggest this:

1. For most use cases go for Routr One as described in the README
2. If you want to use individual components, use the Docker services (with `docker-compose up`)
3. For development, start all the services with `npm start`

Does that help? I won't even suggest installing from source everything is packaged as containers for simplicity.
